dandi/files/zarr.py

```diff
@@ -209,32 +236,43 @@ def get_validation_errors(
         import zarr
 
         errors: list[ValidationResult] = []
+        origin: Origin = Origin(
+            validator=Validator.zarr,
+            validator_version=zarr.__version__,
+            standard=Standard.ZARR,
+            # TODO: standard_version=...,
+        )
+
         try:
             data = zarr.open(str(self.filepath))
-        except Exception:
+        except Exception as e:
             if devel_debug:
                 raise
             errors.append(
                 ValidationResult(
-                    origin=ValidationOrigin(
-                        name="zarr",
-                        version=zarr.version.version,
-                    ),
+                    origin=origin,
                     severity=Severity.ERROR,
                     id="zarr.cannot_open",
                     scope=Scope.FILE,
+                    origin_result=e,
```

Comment:
we might want to adopt later the [CapturedException](https://github.com/datalad/datalad/blob/HEAD/datalad/support/exceptions.py#L25) to avoid dragging too much of context around from the original location of that exception